preptime/routes/users.js

```diff
@@ -0,0 +1,220 @@
+const express = require("express");
+const router = express.Router();
+
+router.get("/", (req, res) => {
+  res.send("USERS API IS RUNNING");
+});
+
+router.get("/userInfo/:userId", async (req, res) => {
+  const admin = req.app.locals.admin;
+  const db = admin.firestore();
+
+  const userId = req.params.userId;
+
+  try {
+    // Get user information from Firestore based on userId
+    const userDoc = await db.collection("users").doc(userId).get();
+
+    if (!userDoc.exists) {
+      return res.status(404).send({ error: "User not found" });
+    }
+
+    const userData = userDoc.data();
+
+    res.send({
+      fullName: userData.fullName || "FULL NAME",
+      profileImg: userData.profileImg || "IMAGE",
+      followers: userData.followers || [],
+      following: userData.followingUsers || [],
+      favoriteTeams: userData.favoriteTeams || [],
+    });
+  } catch (error) {
+    console.error("Error fetching user data:", error);
+    res.status(500).send({ error: "Internal server error" });
+  }
+});
+
+router.post("/followUser", async (req, res) => {
+  const admin = req.app.locals.admin;
+  const db = admin.firestore();
+
+  // Check if req.body is defined
+  if (!req.body) {
+    return res.status(400).send({ error: "Request body is missing" });
+  }
+
+  const { userToFollowId, currentUserId } = req.body;
+
+  if (!userToFollowId || !currentUserId) {
+    return res.status(400).send({ error: "Both userIds are required" });
+  }
+
+  try {
+    // Get references to both user documents
+    const userToFollowRef = db.collection("users").doc(userToFollowId);
+    const currentUserRef = db.collection("users").doc(currentUserId);
+
+    // Get both user documents
+    const [userToFollowDoc, currentUserDoc] = await Promise.all([
+      userToFollowRef.get(),
+      currentUserRef.get(),
+    ]);
+
+    if (!userToFollowDoc.exists || !currentUserDoc.exists) {
+      return res.status(404).send({ error: "One or both users not found" });
+    }
+
+    const userToFollowData = userToFollowDoc.data();
+    const currentUserData = currentUserDoc.data();
+
+    const userToFollowName = userToFollowData.fullName || "Unknown User";
+    const currentUserName = currentUserData.fullName || "Unknown User";
+
+    // Update the documents in a transaction
+    await db.runTransaction(async (transaction) => {
+      // Add currentUser's full name to userToFollow's followers array
+      transaction.update(userToFollowRef, {
+        followers: admin.firestore.FieldValue.arrayUnion(currentUserName),
```

Comment:
Maybe following should be done using the user id instead of name so that it can be unique